### ISSUE TITLE
the Form Item could not edit

### ISSUE THREAD

**Issue by eruca** [ant-design/ant-design#9790]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.3.1

### Environment
macOS 10.13.3 Chrome 63

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/xoq75jm36o)

### Steps to reproduce
the Input can not input any value

### What is expected?
can be edit

### What is actually happening?
not

**Comment by benjycui** [comment]
Please make sure that your demo work.

![image](https://user-images.githubusercontent.com/3580607/37883930-f31439ec-30df-11e8-96b4-977e0259d084.png)

**Comment by eruca** [comment]
sorry, I test it myself by adding something without deleting it. Now it can present the problem;

**Comment by benjycui** [comment]
`BasicComponent` haven't been re-rendered, you can try to simplify your demo and find out why.

**Comment by eruca** [comment]
the BasicComponent add
```js
state = {
    value:''
}
```
and the WrappedComponent add onChange, will work, I change the demo in CodeSanbox;
exists another way without adding state; like
```js
<Form>
{getFieldDecorator(id, option)(<Input />)
</Form>
```

**Comment by eruca** [comment]
I read the code, and found the pro.ant.design Login did not have a state for value,  can you give some hint?

**Comment by eruca** [comment]
@afc163 can help?

**Comment by yesmeck** [comment]
@eruca Your codesandbox seems works now, could you confirm that your issue still exists?

**Comment by eruca** [comment]
yes, but the solution is ugly. I need to add a state useless, just to control re-render, I read the pro.ant.design, there are no state in the BasicComponent, I want to ask why, how can i improve my code ?
